src/__tests__/confidentialDataManager.test.ts

```diff
@@ -16,68 +17,38 @@ describe("confidential data manager tests", () => {
       } as any;
       
 it("decrypt OK from PDV", async () => {
-    jest.spyOn(apiPdvClient, 'findPiiUsingGET').mockResolvedValue({_tag: "Right", right:{ status:200 , value: {pii: JSON.stringify(requestMock.body)}, headers: "" as any} });
-    const result = await pipe(
-        decryptBody(token),
-        TE.map(() => true),
-        TE.mapLeft(() => false),
-        TE.toUnion
-    )()
-   expect(result).toBe(true);
+   jest.spyOn(apiPdvClient, 'findPiiUsingGET').mockResolvedValue({_tag: "Right", right:{ status:200 , value: {pii: JSON.stringify(requestMock.body)}, headers: "" as any} });
+   const result = await decryptBody(token)();
+   expect(E.isRight(result)).toBe(true);
 })
 
 it("decrypt KO from PDV", async () => {
-    jest.spyOn(apiPdvClient, 'findPiiUsingGET').mockResolvedValue({_tag:"Left", left: "" as any });
-    const result = await pipe(
-        decryptBody(token),
-        TE.map(() => true),
-        TE.mapLeft(() => false),
-        TE.toUnion
-    )()
-   expect(result).toBe(false);
+    jest.spyOn(apiPdvClient, 'findPiiUsingGET').mockResolvedValue(E.left([]));
+    const result = await decryptBody(token)();
+    expect(E.isRight(result)).toBe(false);
 })
 
 it("decrypt OK from PDV with ko http status", async () => {
     jest.spyOn(apiPdvClient, 'findPiiUsingGET').mockResolvedValue({_tag: "Right", right:{ status: 403 , value: undefined, headers: "" as any} });
-    const result = await pipe(
-        decryptBody(token),
-        TE.map(() => true),
-        TE.mapLeft(() => false),
-        TE.toUnion
-    )()
-   expect(result).toBe(false);
+    const result = await decryptBody(token)();
+    expect(E.isRight(result)).toBe(false);
 })
 
 it("encrypt OK from PDV", async () => {
     jest.spyOn(apiPdvClient, 'saveUsingPUT').mockResolvedValue({_tag: "Right", right:{ status: 200 , value: {token: token}, headers: "" as any} });
-    const result = await pipe(
-        encryptBody(requestMock.body),
-        TE.map(() => true),
-        TE.mapLeft(() => false),
-        TE.toUnion
-    )()
-   expect(result).toBe(true);
+    const result = await encryptBody(requestMock.body)();
```

Comment:
Here you can avoid constructing a request if you are using only its body

Reply:
Fix with 076486caa8fac19ef44ed4ca2efffdb8ab55cb63)